File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s)!=len(t):
            return False
        counter1,counter2={},{}
        for i in range(len(s)):
            counter1[s[i]]=1+counter1.get(s[i],0)
            counter2[t[i]]=1+counter2.get(t[i],0)
        for i in counter1:
            if counter1[i]!=counter2.get(i,0):
                return False
        return True
        

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // char[] sa=s.toCharArray();
        // char[] ta=t.toCharArray();
        // Arrays.sort(sa);
        // Arrays.sort(ta);
        // return Arrays.equals(sa,ta);
        
        Map<Character,Integer> map=new HashMap<>();
        if(s.length()!=t.length()) return false;
        for(int i=0;i<s.length();i++){
            map.put(s.charAt(i),map.getOrDefault(s.charAt(i),0)+1);
        }
        for(int i=0;i<t.length();i++){
            if(map.containsKey(t.charAt(i))){
                if(map.get(t.charAt(i))==1){
                    map.remove(t.charAt(i));
                }else{
                    map.put(t.charAt(i),map.get(t.charAt(i))-1);
                }
                
            }
        }
        return map.isEmpty();
    }
}
